packages/react-lowcode/src/codegen/generation/generators/list/mui-datatable-generator.ts

```diff
@@ -98,6 +99,80 @@ export default class MuiDataTableGenerator implements TableGenerator
       return alteredSource
     }
 
+    async getColumnSourcePosition(position: SourceLineCol,
+                                  columnIndex: number): Promise<ColumnSourcePositionResult | undefined> {
+      let result: ColumnSourcePositionResult | undefined;
+
+      if (this._widgetContext) {
+        let sourceCode = await this._widgetContext.getSourceCodeString(position)
+
+        let ast = createAst(sourceCode)
+
+        if (ast) {
+          let widgetParentNode = findWidgetParentNode(sourceCode, position)
+
+          if (widgetParentNode) {
+            let columnsDeclarationNode = this.findColumnsDeclaration(widgetParentNode)
+
+            if (columnsDeclarationNode) {
+              let columnDeclarationArray = columnsDeclarationNode.getChildAt(2) as ts.ArrayLiteralExpression
+
+              if (columnDeclarationArray && columnDeclarationArray.elements[columnIndex]) {
```

Comment:
Please, do not repeat `columnDeclarationArray && columnDeclarationArray.elements[columnIndex]`. Modify this code like this:
```
const columnDeclaration = columnDeclarationArray?.elements[columnIndex]
if (columnDeclaration) {
```

Reply:
fixed